mmv1/products/compute/NetworkPeeringRoutesConfig.yaml

```diff
@@ -111,3 +111,19 @@ properties:
     description: |
       Whether to import the custom routes to the peer network.
     send_empty_value: true
+  - !ruby/object:Api::Type::Boolean
+    name: 'exportSubnetRoutesWithPublicIp'
+    description: |
+      Whether subnet routes with public IP range are exported.
+      IPv4 special-use ranges are always exported to peers and
+      are not controlled by this field.
+    send_empty_value: true
+    required: true
+  - !ruby/object:Api::Type::Boolean
+    name: 'importSubnetRoutesWithPublicIp'
+    description: |
+      Whether subnet routes with public IP range are imported.
+      IPv4 special-use ranges are always imported from peers and
+      are not controlled by this field.
+    send_empty_value: true
+    required: true
```

Comment:
I'm concerned that making these fields required is a breaking change to existing deployments. It's why this [testcase](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11491/artifacts/afe31881-f44c-4460-aaaf-114e8b3efe11/build-log/recording_build/TestAccComputeNetworkPeeringRoutesConfig_networkPeeringRoutesConfigGkeExample_recording_test.log) is failing.

Reply:
My concern is that:

1) Private Services Access configures VPC Peering connections with values that differ from the defaults for for [networks.updatePeering](https://cloud.google.com/compute/docs/reference/rest/v1/networks/updatePeering).

   `exportSubnetRoutesWithPublicIp`, `importSubnetRoutesWithPublicIp`, `exportCustomRoutes`, and `importCustomRoutes` are all set to `false` when VPC Peering is created by PSA (at least in my testing with Memorystore for Redis). Whereas the API defaults are `true`, `false`, `false` and `false` respectively.

2) Users may already be managing this resource through Terraform, but have changed `exportSubnetRoutesWithPublicIp` and `importSubnetRoutesWithPublicIp` manually due to lack of TF support.

So I made this required to ensure that users are intentional about setting these values. The danger I can see with setting defaults and making this not required are that we risk users unintentionally changing the values if they do not notice the diff on apply.

The main danger being changing `importSubnetRoutesWithPublicIp` to the default of `false`, which might withdraw routes needed to access services in another VPC and cause production issues.

So I think it might be better to fail safely. But if there is another approach or you disagree LMK.

Reply:
yes, making an optional field required or adding a new required field is strictly a breaking change in Terraform that needs to wait until the next major release.

Reply:
Ok, so I guess the main question is are we in agreement that these fields should be required, without a default based on the rationale above?

If no - what should the alternative be? Make the field default to the API default, and trust users to validate the diffs if they exist?

If yes - whats the process for queuing this PR for the next major release?

Reply:
Hmmm you raise good points ben.  Let me try to figure out whether any Producers in PSA relies on importSubnetRoutesWithPublicIp.  My hunch is that they wouldn't.

Reply:
@kevintchan-goog - I think it depends on the ranges that the customer is using in their VPC, rather than the design on the producer side.

If they are using non RFC-1918 PUPI subnets in their VPC (e.g. [in GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/configuring-privately-used-public-ips-for-gke)) - then traffic can reach the producer VPC, but there are no returns routes exported to the producer VPC, so the traffic is blackholed and the service is inaccessible.

By default, PSA sets `exportSubnetRoutesWithPublicIp` and `importSubnetRoutesWithPublicIp` to `true` on the producer side of the peering, but not on the consumer side.

Reply:
## if we make fields required:

- all users need to update their TF config. This is a mass inconvenience. **How frequent are these types of updates for terraform users?**
- customers risk mistakenly setting a field to wrong value and breaking network connectivity. True/false could both be considered wrong.

**In this case can we omit explicit terraform default values and rely on API defaults instad?**

I'm leaning towards required & either omitted defaults or API-defaults. This seems like the safest option to not get blamed for customer outages.

## If we make fields non-required
- all users need to check that version update doesn't make a secret update to their config. This is not an inconvenience
- customers risk missing an update breaking connectivity. This is an easier mistake to make and blame GCP for.

Reply:
Thanks @kevintchan-goog  - Agree on `default_from_api`. I thought this took the API default, but TIL it will default to the current value given the peering resource already exists. Updated the PR.

Reply:
Hmm - actually no. I am not sure this will work.

Setting `default_from_api` to `true` means that if this is already set to `true` by PSA or the user in the console, setting Terraform to `false` means the field is omitted from the PATCH request and it becomes impossible to set this to `false`.

I tried combining with `send_empty_value` which works well in my local testing but the test case fails and https://googlecloudplatform.github.io/magic-modules/develop/field-reference/#send_empty_value documents:

> send_empty_value and default_from_api cannot both be true on the same field.

So not sure on a path forward.